Lib/test/test_ordered_dict.py

```diff
@@ -970,6 +971,74 @@ def test_weakref_list_is_not_traversed(self):
 
         gc.collect()
 
+    def test_copy_concurrent_clear_in__getitem__(self):
+        class MyOD(self.OrderedDict):
+            def __getitem__(self, key):
+                super().clear()
+                return None
+
+        od = MyOD([(i, i) for i in range(4)])
+        self.assertRaises(RuntimeError, od.copy)
+
+    def test_copy_concurrent_insertion_in__getitem__(self):
+        class MyOD(self.OrderedDict):
+            def __getitem__(self, key):
+                self['new_key'] = 'new_value'
+                return super().__getitem__(key)
+
+        od = MyOD([(1, 'one'), (2, 'two')])
+        self.assertRaises(RuntimeError, od.copy)
+
+    def test_copy_concurrent_deletion_by_del_in__getitem__(self):
+        class MyOD(self.OrderedDict):
+            call_count = 0
+            def __getitem__(self, key):
+                MyOD.call_count += 1
+                if MyOD.call_count == 1:
+                    del self[3]
+                return super().__getitem__(key)
+
+        od = MyOD([(1, 'one'), (2, 'two'), (3, 'three')])
+        self.assertRaises(RuntimeError, od.copy)
+
+    def test_copy_concurrent_deletion_by_pop_in__getitem__(self):
+        class MyOD(self.OrderedDict):
+            call_count = 0
+            def __getitem__(self, key):
+                MyOD.call_count += 1
+                if MyOD.call_count == 1:
+                    self.pop(3, None)
+                return super().__getitem__(key)
+
+        od = MyOD([(1, 'one'), (2, 'two'), (3, 'three')])
+        self.assertRaises(RuntimeError, od.copy)
+
+    def test_copy_concurrent_deletion_and_set_in__getitem__(self):
+        class MyOD(self.OrderedDict):
+            call_count = 0
+            def __getitem__(self, key):
+                MyOD.call_count += 1
+                if MyOD.call_count == 1:
+                    del self[3]
+                elif MyOD.call_count == 2:
+                    self['new_key'] = 'new_value'
+                return super().__getitem__(key)
+
+        od = MyOD([(1, 'one'), (2, 'two'), (3, 'three'), (4, 'four')])
+        self.assertRaises(RuntimeError, od.copy)
+
+    def test_copy_concurrent_mutation_in__setitem__(self):
+        od = None
+
+        class MyOD(self.OrderedDict):
+            def __setitem__(self, key, value):
+                if od is not None and len(od) > 1:
```

Comment:
Use the call count approach. Because this condition could be true if we were to loop again and we would have a keyerror instead here.